patroni/dcs/etcd.py

```diff
@@ -165,7 +165,7 @@ def _get_members(self, base_uri, **kwargs):
     @property
     def machines_cache(self):
         base_uri, cache = self._base_uri, self._machines_cache
-        return ([base_uri] if base_uri in cache else []) + [machine for machine in cache if machine != base_uri]
+        return ([base_uri] if base_uri in cache else []) + list(set([machine for machine in cache if machine != base_uri]))
```

Comment:
IMO, https://github.com/zalando/patroni/blob/256a359a1ecd60ae7b81e1ac6524e1c85d2e78bd/patroni/dcs/etcd.py#L187 is a better place for it.

Reply:
Fair enough, I have changed it